internal/proto/chandata.go

```diff
@@ -58,9 +57,6 @@ func (c *ChannelData) Encode() {
 	c.Raw = c.Raw[:0]
 	c.WriteHeader()
 	c.Raw = append(c.Raw, c.Data...)
-	if !c.Padding {
```

Comment:
Since we don't have a way to distinguish `Stream` vs `Datagram` transport this is easier for now.

We could make this a bool, but I am not sure the complexity is worth it. Happy to re-do if you disagree :)